core/src/main/scala/mongo4cats/client/ClientSession.scala

```diff
@@ -16,72 +16,34 @@
 
 package mongo4cats.client
 
-import cats.Monad
 import cats.effect.Async
-import cats.syntax.alternative._
-import cats.syntax.functor._
+import cats.implicits._
 import com.mongodb.reactivestreams.client.{ClientSession => JClientSession}
 import mongo4cats.helpers._
 
-import scala.util.Try
+final case class ClientSession(session: JClientSession) {
```

Comment:
I think an interface with an impl in the companion would be preferable (and more flexible) to a final class (also doesn't need to be a `case class`).